ocs_ci/ocs/cluster.py

```diff
@@ -426,97 +428,38 @@ def check_ceph_pool_used_space(self, cbp_name):
-            mon_label=label, namespace=defaults.ROOK_CLUSTER_NAMESPACE
-        )
-    if label == constants.DEFAULT_DEVICESET_LABEL:
-        ocs_pod_obj = pod.get_osd_pods(
-            osd_label=constants.OSD_APP_LABEL,
-            namespace=defaults.ROOK_CLUSTER_NAMESPACE
-        )
-
-    # Create a pvc list for requested label
-    pvc_list = []
-    for pvc_obj in ocs_pvc_obj:
-        pvc_list.append(pvc_obj.name)
-
-    # Check if PVC is mounted on designated OCS pod
-    # loop over all PVCs in the pvc_list
-    # Each mon and osd pod is expected to have only one Claim attached
-    claim_found = True
-    all_backed_by_pvc = True
-    for pod_obj in ocs_pod_obj:
-        pod_volumes = pod_obj.get().get('spec').get('volumes')
-        claim_spec_exists = False
-        for volumes in pod_volumes:
-            pvc = volumes.get('persistentVolumeClaim')
-            if pvc:
-                claim_name = pvc.get('claimName')
-                claim_spec_exists = True
-                backed_by_pvc = False
-                if claim_name in pvc_list:
-                    logger.info(
-                        f"OCS pod {pod_obj.name} is backed by PVC {claim_name}"
-                    )
-                    # If backed by PVC, set backed_by_pvc = True
-                    backed_by_pvc = True
-
-                    # Check if Mon PVC is mounted as /var/lib/ceph/mon/ceph-x
-                    if label == constants.MON_APP_LABEL:
-                        mount_point = pod_obj.exec_cmd_on_pod(command="df -kh")
-                        assert "/var/lib/ceph/mon/ceph" in mount_point, (
-                            f"pvc is not mounted on pod {pod_obj.name}"
-                        )
-                        logger.info(
-                            f"PVC {claim_name} is mounted"
-                            f" on pod {pod_obj.name}"
-                        )
-        if not backed_by_pvc:
-            logger.error(f"{pod_obj.name} is not backed by designated PVC ")
-            all_backed_by_pvc = False
-
-        # If no PVC is mounted, print error and continue checking other pods
-        if not claim_spec_exists:
-            logger.error(
-                f"No PersistentVolumeClaim spec found in "
-                f"OCS pod {pod_obj.name}"
-            )
-            claim_found = False
-
-    # Even if one OCS POD is not backed by a PVC, fail the deployment
-    assert all_backed_by_pvc, (
-        "One or more pods are not backed by a PVC "
-        "please check deployment logs"
-    )
-
-    # Even if one OCS POD as PVC spec, fail the deployment
-    assert claim_found, (
-        "Claim name doesn't exists in spec of one or more pods "
-        "please check deployment logs"
-    )
+        pvc_names.append(pvc_obj.name)
+
+    mon_pods = get_pod_name_by_pattern('rook-ceph-mon', ns)
+    osd_pods = get_pod_name_by_pattern('rook-ceph-osd', ns, filter='prepare')
+    assert len(mon_pods) + len(osd_pods) == len(pvc_names), \
```

Comment:
Please do not use backslashes and rather split line using brackets as it's our code convention.